itou/www/apply/tests/tests_submit.py

```diff
@@ -104,6 +105,48 @@ def test_start_coalesce_back_url(self):
         assert self.client.session[f"job_application-{siae.pk}"]["back_url"] == "/"
 
 
+def test_check_nir_job_seeker_with_lack_of_nir_reason(client):
+    """Apply as jobseeker."""
+
+    siae = SiaeWithMembershipAndJobsFactory(romes=("N1101", "N1105"))
+
+    user = JobSeekerFactory(birthdate=None, nir=None, lack_of_nir_reason=LackOfNIRReason.TEMPORARY_NUMBER)
+    client.force_login(user)
+
+    # Entry point.
+    # ----------------------------------------------------------------------
+
+    response = client.get(reverse("apply:start", kwargs={"siae_pk": siae.pk}), {"back_url": "/"})
+    assert response.status_code == 302
+
+    session_data = client.session[f"job_application-{siae.pk}"]
+    assert session_data == {
+        "back_url": "/",
+        "job_seeker_pk": user.pk,
+        "nir": None,
+        "selected_jobs": [],
+    }
+
+    next_url = reverse("apply:check_nir_for_job_seeker", kwargs={"siae_pk": siae.pk})
+    assert response.url == next_url
+
+    # Step check job seeker NIR.
+    # ----------------------------------------------------------------------
+
+    response = client.get(next_url)
+    assert response.status_code == 200
+
+    nir = "141068078200557"
+    post_data = {"nir": nir, "confirm": 1}
+
+    response = client.post(next_url, data=post_data)
+    assert response.status_code == 302
+
+    user = User.objects.get(pk=user.pk)
```

Comment:
Comme tu préfères, c’est surtout pour la lisibilité.
```suggestion
    user.refresh_from_db()
```